File: latihan2/cpp/main.cpp
#include <string>
#include <iostream>
#include <bits/stdc++.h>
#include "product.cpp"
#include "hardware.cpp"
#include "memory.cpp"

using namespace std;

int main(){

    // semua instansiasi dilakukan dgn method setter

    product produk;

    // deklarasi variabel
    int harga, frekuensi, ukuran;
    string id, merk, bentuk, cuda;

    // input
    cin >> harga >> id;

    produk.setPrice(harga);
    produk.setIdProduct(id);

    // output dgn getter
    cout << "==============================================" << endl;
    cout << "Harga : Rp" << produk.getPrice() << endl; 
    cout << "ID Produk : " << produk.getIdProduct() << endl; 

    hardware keras;

    cin >> merk >> bentuk;

    // child mengakses method parent product
    keras.setPrice(harga);
    keras.setIdProduct(id);
    keras.setBrand(merk);
    keras.setModel(bentuk);

    cout << "==============================================" << endl;
    cout << "Merek : " << keras.getBrand() << endl; 
    cout << "Model : " << keras.getModel() << endl; 
    cout << "Harga : Rp" << keras.getPrice() << endl; 
    cout << "ID Produk : " << keras.getIdProduct() << endl; 

    memory memori;

    cin >> frekuensi >> ukuran >> cuda;

    // child mengakses method parent product dan hardware
    memori.setPrice(harga);
    memori.setIdProduct(id);
    memori.setBrand(merk);
    memori.setModel(bentuk);
    memori.setFrequency(frekuensi);
    memori.setMemorySize(ukuran);
    memori.setSupportCuda(cuda);

    cout << "==============================================" << endl;
    cout << "Merek : " << memori.getBrand() << endl; 
    cout << "Model : " << memori.getModel() << endl; 
    cout << "ID Produk : " << memori.getIdProduct() << endl; 
    cout << "Frekuensi : " << memori.getFrequency() << "MHz" << endl; 
    cout << "Ukuran Memory : " << memori.getMemorySize() << "GB" << endl; 
    cout << "Support Cuda :" << memori.getMemorySize() << endl; 
    cout << "Harga : Rp" << memori.getPrice() << endl; 

    return 0;
}

File: latihan2/java/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {

        // Semua instansiasi menggunakan metode setter

        product produk = new product();

        // deklarasi variabel
        int harga, frekuensi, ukuran;
        String id, merek, bentuk, cuda;

        // instansiai objek Scanner untuk input
        Scanner sc = new Scanner(System.in);
        
        // input untuk class product (orang tua)
        harga = sc.nextInt();
        id = sc.next();
        
        // memasukkan input melalui setter
        produk.setPrice(harga);
        produk.setIdProduct(id);
        
        // menampilkan output dgn metode getter untuk class parent
        System.out.println("==============================================");
        System.out.println("Harga : Rp" + produk.getPrice());        
        System.out.println("ID Produk : " + produk.getIdProduct());        
        
        hardware keras = new hardware();
        
        // memasukkan input melalui setter
        merek = sc.next();
        bentuk = sc.next();
        
        keras.setBrand(merek);
        keras.setModel(bentuk);
        keras.setPrice(harga);
        keras.setIdProduct(id);
        
        // menampilkan output dgn metode getter untuk class child
        System.out.println("==============================================");
        System.out.println("Merek : " + keras.getBrand());        
        System.out.println("Model : " + keras.getModel());        
        System.out.println("Harga : Rp" + keras.getPrice());        
        System.out.println("ID Produk : " + keras.getIdProduct());  
        
        memory chip = new memory();
        
        // memasukkan input melalui setter
        frekuensi = sc.nextInt();
        ukuran = sc.nextInt();
        cuda = sc.next();
        
        chip.setFrequency(frekuensi);
        chip.setMemorySize(ukuran);
        chip.setSupportsCuda(cuda);
        chip.setBrand(merek);
        chip.setModel(bentuk);
        chip.setPrice(harga);
        chip.setIdProduct(id);
        
        // menampilkan output dgn metode getter untuk class child
        System.out.println("==============================================");
        System.out.println("Merek : " + chip.getBrand());        
        System.out.println("Model : " + chip.getModel());        
        System.out.println("ID Produk : " + chip.getIdProduct());  
        System.out.println("Frekuensi : " + chip.getFrequency() + "MHz");        
        System.out.println("Ukuran Memory : " + chip.getMemorySize() + "GB");        
        System.out.println("Support Cuda : " + chip.getSupportsCuda());        
        System.out.println("Harga : Rp" + chip.getPrice());        

    }
}
// bola player1 = new bola();
// player1.setNama("Aespa");
// player1.setNegara("Kroya");
// player1.setTahun(2013);
// player1.setPemain("Karina");